.ci/release/Jenkinsfile

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/env groovy
+
+@Library('apm@current') _
+pipeline {
+  agent { label 'linux && immutable' }
+  environment {
+    REPO = 'apm-agent-java'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    NOTIFY_TO = credentials('notify-to')
+    DOCKERHUB_SECRET = 'secret/apm-team/ci/elastic-observability-dockerhub'
+    ELASTIC_DOCKER_SECRET = 'secret/apm-team/ci/docker-registry/prod'
+    NEXUS_SECRET = 'secret/apm-team/ci/nexus'
+    MAVEN_CONFIG = '-B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -Dmaven.repo.local=.m2'
+    HOME = "${env.WORKSPACE}"
+    JAVA_HOME = "${env.HUDSON_HOME}/.java/java10"
+    PATH = "${env.JAVA_HOME}/bin:${env.PATH}"
+  }
+  options {
+    timeout(time: 3, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+
+  }
+  parameters {
+    string(name: 'branch_specifier', defaultValue: 'master')
+  }
+
+  stages {
+    stage('Initializing'){
+      options { skipDefaultCheckout() }
+      stages {
+        /**
+         Checkout the code
+        */
+        stage('Checkout') {
+          steps {
+            dir("${BASE_DIR}") {
+              // This is required to ensure that Maven is working with a real branch
+              sh(script: "git checkout -f ${branch_specifier}")
```

Comment:
A few comments regarding the git commands
- Is it possible to use the `gitCheckout` step instead the `sh` step? If so, then we can remove the `githubEnv` references 
- Can we use the `withGitRelease` step before pushing with the command `gitPush`?
- Can we use the `setupAPMGitEmail` step to configure the user.email and user.name?

Reply:
I moved this into a shell script. Could you take another look?